packages/core/content-manager/server/controllers/collection-types.js

```diff
@@ -46,16 +52,26 @@ module.exports = {
       return ctx.forbidden();
     }
 
-    const entity = await entityManager.findOneWithCreatorRolesAndCount(id, model);
+    const permissionQuery = await permissionChecker.sanitizedQuery.read(ctx.query);
+    const populate = await populateBuilder(model)
+      .populateFromQuery(permissionQuery)
+      .populateDeep(Infinity)
+      .countRelations()
+      .build();
+
+    const entity = await entityManager.findOne(id, model, { populate });
 
     if (!entity) {
       return ctx.notFound();
     }
 
+    // if the user has condition that needs populated content, it's not applied because entity don't have relations populated
     if (permissionChecker.cannot.read(entity)) {
       return ctx.forbidden();
     }
 
+    // TODO: Count populated relations by permissions
```

Comment:
For now this is not necessary and the only downside of not returning counted relations is the relation count flickers when loading relations.